src/app/pages/TopicPage/Pagination/index.jsx

```diff
@@ -0,0 +1,173 @@
+  margin-right: 0.25rem;
+  &:last-of-type {
+    margin-right: 0;
+  }
+  svg {
+    fill: currentColor;
+    width: 1rem;
+    height: 1rem;
+    position: relative;
+    top: 0.2rem;
+  }
+`;
+
+const ActiveBlock = styled(Block)`
+  color: ${C_WHITE};
+  background: #696969;
+`;
+const UnavailableBlock = styled(Block)`
+  color: #bababa;
+`;
+const EllipsisBlock = styled(Block)`
+  color: #8a8c8e;
+  background: unset;
+`;
+
+const selectWrapper = availability =>
+  ({
+    [AVAILABILITY.ACTIVE]: ActiveBlock,
+    [AVAILABILITY.UNAVAILABLE]: UnavailableBlock,
+    [AVAILABILITY.AVAILABLE]: Block,
+  }[availability]);
+
+const A = styled.a`
+  display: block;
+  color: ${C_EBON};
+  text-decoration: none;
+  height: 100%;
+  width: 100%;
+  &:hover {
+    color: white;
+    background: #5a5a5a;
+  }
+`;
+
+/* eslint-disable react/prop-types */
+const LinkComponent = ({ children, availability, pageNumber }) => {
+  if (availability === AVAILABILITY.AVAILABLE) {
+    return <A href={`?page=${pageNumber}`}>{children}</A>;
+  }
+  return children;
+};
+
+const blockComponents = {
+  [TYPE.NUMBER]: ({ availability, pageNumber, service }) => {
+    const BlockComponent = selectWrapper(availability);
+    return (
+      <BlockComponent service={service}>
+        <LinkComponent availability={availability} pageNumber={pageNumber}>
+          {pageNumber}
+        </LinkComponent>
+      </BlockComponent>
+    );
+  },
+  [TYPE.LEFT_ARROW]: ({ availability, activePage }) => {
+    const BlockComponent = selectWrapper(availability);
+    return (
+      <BlockComponent>
+        <LinkComponent availability={availability} pageNumber={activePage - 1}>
+          <LeftChevron />
+        </LinkComponent>
+      </BlockComponent>
+    );
+  },
+  [TYPE.RIGHT_ARROW]: ({ availability, activePage }) => {
+    const BlockComponent = selectWrapper(availability);
+    return (
+      <BlockComponent>
+        <LinkComponent availability={availability} pageNumber={activePage + 1}>
+          <RightChevron />
+        </LinkComponent>
+      </BlockComponent>
+    );
+  },
+  [TYPE.ELLIPSIS]: () => (
+    <EllipsisBlock>
+      <Ellipsis />
+    </EllipsisBlock>
+  ),
+};
+/* eslint-enable react/prop-types */
+
+const Pagination = ({ activePage, pageCount }) => {
+  const { service } = useContext(ServiceContext);
+  const clampedPageCount = clamp(1, 99, pageCount);
+  const clampedActivePage = clamp(1, clampedPageCount, activePage);
+  const blocks = buildBlocks(clampedActivePage, clampedPageCount);
+  if (!blocks) return null;
+
+  return (
+    <OL role="list">
+      {blocks.map(block => {
+        const Component = blockComponents[block.type];
+        return (
+          <Component {...block} activePage={activePage} service={service} />
+        );
+      })}
+    </OL>
+  );
+};
+
+Pagination.propTypes = {
+  activePage: number,
+  pageCount: number,
+};
+
+Pagination.defaultProps = {
+  activePage: 1,
+  pageCount: 1,
+};
+
+export default Pagination;
```

Comment:
This component needs some tests to show it can take varying page ranges and return suitable visual components. It's not about retesting `buildBlocks` as that has a lot of the complex view logic, but checking that the correct dom elements are rendered for different page ranges. It doesn't need to be as comprehensive as the `buildBlocks` test.